### PR TITLE
[WIP] / [DRAFT] / [ETC] implement uri method

### LINKED ISSUE #278: `:uri` attribute
a `:uri` attribute can automatically be transformed to `href="real URI"`

```
<a :uri="HomeController::class"></a>
<a href="/"></a>
```

### PR DESCRIPTION
**Pull request by Treggats**
resolves #278

mostly here so that I have a nice overview of my changes.

### CODE REVIEW

File: src/Tempest/Console/Commands/DiscoveryClearCommand.php [hunk 1]
```diff
@@ -31,6 +31,10 @@ public function __invoke(): void
 
             $discovery->destroyCache();
 
+            if ($this->appConfig->environment->isTesting()) {
```
Comment (Treggats) [inline src/Tempest/Console/Commands/DiscoveryClearCommand.php:34]: the messages aren't really needed during testing, so skipping them.

File: src/Tempest/View/Attributes/UriAttribute.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+use Tempest\Container\Container;
+use Tempest\Container\GenericContainer;
+use Tempest\Http\Get;
+use Tempest\Http\Post;
+use Tempest\View\Attribute;
+use Tempest\View\Element;
+use Tempest\View\Elements\GenericElement;
+use Tempest\View\Elements\TextElement;
+
+final class UriAttribute implements Attribute
```
Comment (Treggats) [inline src/Tempest/View/Attributes/UriAttribute.php:20]: new attribute for the `:uri` attribute. a very WIP at the moment though

File: tests/Integration/FrameworkIntegrationTestCase.php [hunk 3]
```diff
@@ -28,6 +29,7 @@ protected function setUp(): void
     {
         $this->appConfig = new AppConfig(
             root: __DIR__ . '/../../',
+            environment: Environment::TESTING,
```
Comment (Treggats) [inline tests/Integration/FrameworkIntegrationTestCase.php:32]: was set to local before, but in here we're testing. so set the environment accordingly

File: tests/Integration/View/TempestViewRendererTest.php [hunk 4]
```diff
@@ -49,6 +52,38 @@ public function test_if_attribute()
...
+    public function it_can_render_an_a_element_with_href_attribute(string $expected, string $action, string $body): void
+    {
+        $this->markTestSkipped();
+        $this->assertSame(
+            $expected,
+            $this->render(view(sprintf('<a :uri="%s">%s</a>', $action, $body))),
+        );
+    }
+
+    public static function provide_a_attribute_values(): Generator
```
Comment (Treggats) [inline tests/Integration/View/TempestViewRendererTest.php:66]: provide different formats to test with

**Comment by brendt** [comment]
Gonna close this one for now, I think it's too complicated with how we need to pass in variables. Thanks for looking into it!

**Pull request closed without merging**